tap_shopify/streams/metafields.py

```diff
@@ -1,82 +1,182 @@
+from datetime import timedelta
 import json
 import shopify
-import singer
+
+from singer import utils, get_logger
 
 from tap_shopify.context import Context
-from tap_shopify.streams.base import (Stream,
-                                      shopify_error_handling,
-                                      RESULTS_PER_PAGE,
-                                      OutOfOrderIdsError)
-
-LOGGER = singer.get_logger()
-
-def get_selected_parents():
-    for parent_stream in ['orders', 'customers', 'products', 'custom_collections']:
-        if Context.is_selected(parent_stream):
-            yield Context.stream_objects[parent_stream]()
-
-@shopify_error_handling
-def get_metafields(parent_object, since_id, parent_replication_object, timeout):
-    # set timeout
-    parent_replication_object.set_timeout(timeout)
-    # This call results in an HTTP request - the parent object never has a
-    # cache of this data so we have to issue that request.
-    return parent_object.metafields(
-        limit=Context.get_results_per_page(RESULTS_PER_PAGE),
-        since_id=since_id)
-
-class Metafields(Stream):
+from tap_shopify.streams.graphql import (
+    get_parent_ids_query,
+    get_metafield_query_customers,
+    get_metafield_query_product,
+    get_metafield_query_collection,
+    get_metafield_query_order,
+    get_metafield_query_shop,
+)
+from tap_shopify.streams.graphql.gql_base import (
+    ShopifyGqlStream,
+    ShopifyGraphQLError,
+    DATE_WINDOW_SIZE,
+    shopify_error_handling,
+    )
+
+
+LOGGER = get_logger()
+
+
+
+class Metafields(ShopifyGqlStream):
     name = 'metafields'
-    replication_object = shopify.Metafield
+    data_key = "metafields"
+    replication_key = "updatedAt"
+
+    selected_parent = None
+
+    parent_alias = {
+        "custom_collections":"collections"
+    }
+
+
+    # maps object list identifier to single object access identifier
+    # eg customers -> customer
+    resource_alias = {
+        "customers":"customer",
+        "products":"product",
+        "collections": "collection",
+        "orders": "order"
+    }
```

Comment:
Minor fix, won't block the PR but get it fixed.

```suggestion
    resource_alias = {
        "customers": "customer",
        "products": "product",
        "collections": "collection",
        "orders": "order"
    }
```